receptai.api/Controllers/RecipeVoteController.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using receptai.api.Dtos.RecipeVote;
+using receptai.api.Interfaces;
+using receptai.api.Mappers;
+using receptai.api.Repositories;
+using receptai.data;
+
+namespace receptai.api.Controllers;
+
+[Route("api/vote")]
+[ApiController]
+public class RecipeVoteController : ControllerBase
+{
+    private readonly IRecipeVoteRepository _recipeVoteRepository;
+    private readonly IRecipeRepository _recipeRepository;
+
+    public RecipeVoteController(
+        IRecipeVoteRepository recipeVoteRepository,
+        IRecipeRepository recipeRepository)
+    {
+        _recipeVoteRepository = recipeVoteRepository;
+        _recipeRepository = recipeRepository;
+    }
+
+    [HttpGet("{userId}/{recipeId}")]
+    public async Task<IActionResult> GetByUserAndRecipeId(
+        [FromRoute] int userId, [FromRoute] int recipeId)
+    {
+        var recipeVote = await _recipeVoteRepository.GetByUserAndRecipeId(
+            userId, recipeId);
+
+        if (recipeVote is null)
+        {
+            return NotFound();
+        }
+
+        return Ok(recipeVote.ToRecipeVoteDto());
+    }
+
+    [HttpGet("{recipeId}")]
+    public async Task<IActionResult> GetByRecipeId([FromRoute] int recipeId)
+    {
+        var recipe = await _recipeRepository.GetByIdAsync(recipeId);
+
+        if (recipe is null)
+        {
+            return NotFound();
+        }
+
+        var recipeVotes = await _recipeVoteRepository.GetByRecipeId(recipeId);
+
+        return Ok(recipeVotes.Select(rv => rv.ToRecipeVoteDto()).ToList());
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> Create(
+        [FromBody] CreateRecipeVoteRequestDto recipeVoteDto)
+    {
```

Comment:
should use [Authorize], user id taken from claims principal via User.GetId();

Reply:
same goes for other endpoints which take in user id and relate to doing actions "for" the user